File: Codechef/May/Cook/Phatasiland.cpp
#include <bits/stdc++.h>
#define vi vector<int>
#define all(v) v.begin(),v.end()
#define pii pair<int,int>
#define mp make_pair
#define ll long long
using namespace std;
int arr[100007];
int main() {
#ifndef ONLINE_JUDGE
	freopen("/home/shiva/Learning/1.txt", "r", stdin);
	freopen("/home/shiva/Learning/2.txt", "w", stdout);
#endif
	ll n; ll x; cin>>n;
	ll total = 0;
	for(int i=0;i<n;i++) {
		cin>>x; total+=x;
	}
	cout<<max(n,(total+1)/2)<<endl;
}

File: Codechef/May/Cook/Phatasiland.py
from sys import stdin,stdout
# stdin = open("/home/shiva/Learning/1.txt","r")
# stdout = open("/home/shiva/Learning/2.txt","w")
def main():
	t = int(stdin.readline().strip())
	total = sum((i for i in map(int,stdin.readline().strip().split())))
	stdout.write(str(max(t,int((total+1)/2))))
main()